### PR TITLE
[FIX] Checks for text before newlines on DVB subtitles

### PR DESCRIPTION
**Pull request by gray-v**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [X] I have used CCExtractor just a couple of times.

---

Resolves the multiple lines issue of https://github.com/CCExtractor/ccextractor/issues/840, basically does the same thing that https://github.com/CCExtractor/ccextractor/pull/844 does but without raising potential issues with newlines as pointed out in the codein task.

### CODE REVIEW

**Comment by harrynull** [comment]
Did you try the program without -nodvbcolor? because dvb color will also break new line. Also, did you consider UTF8 characters when you check for ASCII characters?
Another suggestion is, although I am not sure if it will happen, what if the line only contains 0/1 character?

**Comment by gray-v** [comment]
Yeah ideally I'd like to just check the entire line by scanning backwards to the previous \n or start of message and then checking if there is at least 1 letter of some sort in it but I couldn't quite get it to work because I can't find what marks the actual start of the subtitle line, as I got an infinite loop where it just rolled back and back and back forever.

**Comment by gray-v** [comment]
after discussion on slack wait for #844 to be merged, then I'll update this and put it on top of that

File: src/lib_ccx/ocr.c [hunk 1]
```diff
@@ -296,15 +298,6 @@ char* ocr_bitmap(void* arg, png_color *palette,png_byte *alpha, unsigned char* i
 				int x1, y1, x2, y2;
 				if (!TessPageIteratorBoundingBox((TessPageIterator *)ri, level, &x1, &y1, &x2, &y2))
 					continue;
-				// printf("word: '%s';  \tconf: %.2f; BoundingBox: %d,%d,%d,%d;",word, conf, x1, y1, x2, y2);
```
Comment (cfsmp3) [inline src/lib_ccx/ocr.c]: Not sure we want to completely remove this. Seems like code that whoever is doing maintenance/debugging may want to enable from time to time.

File: src/lib_ccx/ocr.c [hunk 2]
```diff
@@ -715,8 +707,8 @@ static int quantize_map(png_byte *alpha, png_color *palette,
 			int index = iot[mcit[mxi]];
 			alpha[iot[i]] = alpha[index];
 			palette[iot[i]].red = palette[index].red;
-			palette[iot[i]].blue = palette[index].blue;
-			palette[iot[i]].green = palette[index].green;
+palette[iot[i]].blue = palette[index].blue;
```
Comment (cfsmp3) [inline src/lib_ccx/ocr.c]: This is breaking indent.

File: src/lib_ccx/ocr.c [hunk 3]
```diff
@@ -728,51 +720,52 @@ static int quantize_map(png_byte *alpha, png_color *palette,
 			i, palette[i].red, palette[i].green, palette[i].blue, alpha[i]);
 	}
 #endif
-	end: freep(&histogram);
+end: freep(&histogram);
```
Comment (cfsmp3) [inline src/lib_ccx/ocr.c]: Breaking indent.

File: src/lib_ccx/ocr.c [hunk 4]
```diff
@@ -810,14 +803,32 @@ int compare_rect_by_ypos(const void*p1, const void *p2, void*arg)
 
 void add_ocrtext2str(char *dest, char *src, const char *crlf, unsigned crlf_length)
 {
+	char *line_scan;
+	BOOL should_break = FALSE;
```
Comment (cfsmp3) [inline src/lib_ccx/ocr.c]: I'm not a big fan of BOOL - in C bools are int, 0 is FALSE and everything else is TRUE...

File: src/lib_ccx/ocr.c (hunk 1 above)
Comment (gray-v) [reply src/lib_ccx/ocr.c]: Fair point, but printf literally doesn't even work here. I'll change it to a mprint()

**Pull request closed without merging**